### ISSUE TITLE
[4.0] [a11y] View categories needs improvement

### ISSUE THREAD

**Issue by chmst** [joomla/joomla-cms#33037]
### Steps to reproduce the issue
Make a menuItem to a category List.

The result looks like this: the Category Help has a subcategory but the button for expanding is hardly visible. 
All items (including the badges) are tagged as h3, which is wrong in most cases not only accessibility
The tree sstructure is not visible at all. 

### Expected result
A propery looking tree structure with buttons for expand/collapse near to the respective category.

### Actual result
![grafik](https://user-images.githubusercontent.com/1035262/113713095-36723c00-96e7-11eb-8ae4-57c9a1587dcf.png)

**Comment by brianteeman** [comment]
Everything is inside a single h3 🤦

**Comment by drmenzelit** [comment]
Closing as we have a PR #33052
